Library/Homebrew/cask/cask.rb

```diff
@@ -170,6 +170,8 @@ def outdated_versions(greedy: false, greedy_latest: false, greedy_auto_updates:
         if latest_version.latest?
           return versions if outdated_download_sha?
 
+          return []
+        elsif greedy_latest && auto_updates
```

Comment:
```suggestion
        elsif !(greedy || greed_auto_update) && auto_updates
```
In theory, multiple greedy flags could be set and this should cover all of them.

Reply:
@apainintheneck Makes sense, thanks.